src/math/p5.Vector.js

```diff
@@ -2138,6 +2151,34 @@ p5.Vector.mult = function mult(v, n, target) {
   return target;
 };
 
+/**
+ * Rotates the vector (only 2D vectors) by the given angle, magnitude remains the same and returns a new vector.
+ */
+
+/**
+ * @method rotate
+ * @static
+ * @param  {p5.Vector} v
+ * @param  {Number} angle
+ * @param  {p5.Vector} [target] the vector to receive the result (Optional)
+ */
+
+p5.Vector.rotate = function rotate(v, a, target) {
+  if (!target) {
+    target = v.copy();
+    if (arguments.length === 3) {
```

Comment:
Would it make sense to put this check first? I'm getting a little confused reading this in the flow here.

Reply:
I added the check here since other static functions having a `target` variable follows the same fashion. Here are those functions [mult](https://github.com/processing/p5.js/blob/main/src/math/p5.Vector.js#L2126), [div](https://github.com/processing/p5.js/blob/main/src/math/p5.Vector.js#L2126), [lerp](https://github.com/processing/p5.js/blob/main/src/math/p5.Vector.js#L2251). I also feel that it would be better to have the checks later on (in the else statement). Shall I make the change?

Reply:
Ah I see. I think so. @limzykenneth do you see any reason it would be written in this order?

Reply:
No particular reason I can see, it is better in the `else` block. The checks could be better written though with the first `if(!target)` check be an arguments length check while the friendly error check be an `instanceof p5.Vector` check in the `else` block.

Reply:
I have made some changes to the checks. Kindly review it.
Thanks

Reply:
The new change isn't quite what I mean. It's a bit redundant now with two `copy()`. The friendly error check can just be in the else block of the argument check (ie. nested) without being a separate block.

Reply:
That's closer

Reply:
I moved the negation outside the brackets and updated the new changes. How does it look now?

Reply:
I'm not 100% sure about the wording of the friendly error (can maybe be even friendlier?) but other than that, the implementation is more or less what I have in mind.

Reply:
I came up with these three possible error statements.
1. Can't assign the resultant vector to the target parameter. It should be of type p5.Vector
2. The target parameter is an invalid 'instanceof' p5.Vector. It should be of type p5.Vector
3. Invalid assignment to the target parameter. It should be of type p5.Vector

The first statement feels friendlier out of the three. We could also replace the 'resultant vector' in the first statement with 'rotated vector', 'vector', or 'vector object'. Thoughts on this?

Reply:
Actually I think it's fine for now since the other functions uses similar wording as well.